**Soal 1. Data Preparation**

Bacalah Dataset dengan memberikan beberapa argumen sehingga bisa membaca datetime

In [6]:
import pandas as pd
import datetime

In [7]:
df = pd.read_csv('climate_change.csv', index_col = 'date', parse_dates = True)
df.head(5)

meantemp   humidity  wind_speed  meanpressure
date                                                      
2013-01-01  10.000000  84.500000    0.000000   1015.666667
2013-01-02   7.400000  92.000000    2.980000   1017.800000
2013-01-03   7.166667  87.000000    4.633333   1018.666667
2013-01-04   8.666667  71.333333    1.233333   1017.166667
2013-01-05   6.000000  86.833333    3.700000   1016.500000

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1TqrXs9JStRXqiF0NIWiMdo5r8CQ-S74-)



---



---



Soal 2. Selection

Tampilkan data keseluruhan untuk tahun 2015 di bulan November

In [10]:
#code here
df['2015-11'].head(7)

C:\Users\ian\AppData\Local\Temp\ipykernel_14564\1121414022.py:2: FutureWarning: Indexing a DataFrame with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. Use `frame.loc[string]` instead.
  df['2015-11'].head(7)


meantemp  humidity  wind_speed  meanpressure
date                                                    
2015-11-01    22.875    70.500      0.9250      1016.250
2015-11-02    23.375    71.375      0.4625      1015.875
2015-11-03    24.000    67.875      0.4750      1014.250
2015-11-04    23.750    69.000      0.9250      1012.250
2015-11-05    22.125    75.250      5.1000      1012.375
2015-11-06    22.875    74.375      2.3250      1012.750
2015-11-07    21.875    74.875      1.1625      1014.625

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1ABMLSmdQ817PIdxW1v7rUKelOj_W241g)



---



---



**Soal 3. Filtering**

Tampilkan data perubahan iklim pada 1 bulan terakhir di tahun 2016 dan jawablah pertanyaan berikut:

Berapakah rata-rata temperatur tertinggi?

In [13]:
#code here
df['2016'].last('1M').head(7)

C:\Users\ian\AppData\Local\Temp\ipykernel_14564\1080001424.py:2: FutureWarning: Indexing a DataFrame with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. Use `frame.loc[string]` instead.
  df['2016'].last('1M').head(7)


meantemp   humidity  wind_speed  meanpressure
date                                                      
2016-12-01  19.750000  84.000000    2.704167   1016.875000
2016-12-02  19.208333  75.875000    4.945833   1017.750000
2016-12-03  21.208333  52.166667    5.866667   1019.333333
2016-12-04  18.900000  55.250000    5.666667   1019.700000
2016-12-05  18.636364  56.590909    4.952381   1017.045455
2016-12-06  18.538462  69.923077    2.503846   1017.961538
2016-12-07  18.250000  74.350000    0.925000   1017.421053

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1X2hXsvrRGvqHqtc5BFbVvRXwh5YxBQ_P)

In [15]:
df['2016']['meantemp'].last('1M').max()

C:\Users\ian\AppData\Local\Temp\ipykernel_14564\1555422294.py:1: FutureWarning: Indexing a DataFrame with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. Use `frame.loc[string]` instead.
  df['2016']['meantemp'].last('1M').max()


21.20833333

Jawab disini:

21.20833333

**Soal 4. Time grouping**

 

*   Lakukan grouping data perubahan iklim pada tahun 2016 berdasarkan frekuensi 1 bulan dengan memberikan agregasi max pada kolom meantemp. humidity, wind_speed dan meanpressure!
*   Berikan analisa, di bulan apa rata-rata temperatur, humidity, wind_speed dan meanpressure tertinggi?





In [25]:
#code here
df_freq1 = df['2016'].groupby(pd.Grouper(freq='1M')).max()
df_freq1

C:\Users\ian\AppData\Local\Temp\ipykernel_14564\282757663.py:2: FutureWarning: Indexing a DataFrame with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. Use `frame.loc[string]` instead.
  df_freq1 = df['2016'].groupby(pd.Grouper(freq='1M')).max()


meantemp   humidity  wind_speed  meanpressure
date                                                      
2016-01-31  20.142857  88.833333   11.240000   1021.142857
2016-02-29  23.875000  77.000000   17.137500   1020.071429
2016-03-31  30.000000  67.800000   22.100000   7679.333333
2016-04-30  35.687500  44.250000   19.653846   1009.400000
2016-05-31  38.272727  74.812500   18.425000   1007.466667
2016-06-30  38.200000  75.071429   21.306250   1002.833333
2016-07-31  34.187500  93.875000   13.656250   1004.625000
2016-08-31  33.809524  89.555556   14.666667   1006.200000
2016-09-30  33.444444  87.240000   17.908333   1352.615385
2016-10-31  33.269231  71.875000   15.825000   1017.807692
2016-11-30  25.640000  80.086957   15.085000   1350.296296
2016-12-31  21.208333  94.300000   19.404762   1019.700000

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1s2Y8Dc3M3xbG-51mTuTmXUFgLvHDvc5Q)

In [26]:
df_freq1.max()

meantemp          38.272727
humidity          94.300000
wind_speed        22.100000
meanpressure    7679.333333
dtype: float64

Jawab disini:

temp tertinggi = april
humidity tertinggi = desember
wind speed tertinggi = maret
pressure tertinggi = maret

**Soal 5. Time Differenced**

Tunjukkanlah perubahan iklim di tahun 2016 pada minggu terakhir

In [28]:
#code here
df['2016'].diff().last('1W')

C:\Users\ian\AppData\Local\Temp\ipykernel_14564\3002856194.py:2: FutureWarning: Indexing a DataFrame with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. Use `frame.loc[string]` instead.
  df['2016'].diff().last('1W')


meantemp   humidity  wind_speed  meanpressure
date                                                     
2016-12-26  3.142857 -19.442857   -0.300789      2.602381
2016-12-27 -0.292857  -7.307143   -0.449211      0.247619
2016-12-28  0.367391   0.493478   -4.787174     -1.634783
2016-12-29 -1.979296  19.813665    2.452174      1.339545
2016-12-30 -1.142857   1.809524    0.266667      1.000000
2016-12-31  0.957393  -2.666667    1.058333     -1.804762

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1miJBQeGMyUS_G3bwP14Plo4s5950oYy5)



---



---

